### PR TITLE
Fix access token behavior in connection pool

### LINKED ISSUE #438: Connection pool doesn't work well with access tokens
### Describe the bug
The connection pool doesn't work well with explicit access tokens because SqlConnectionPoolKey uses reference equality to compare access tokens. It should use string.Equals instead.

The error is in line https://github.com/dotnet/SqlClient/blob/654933a0811ab30b843a20aff051b8d585238505/src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/SqlConnectionPoolKey.cs#L113

### To reproduce
Create two strings at1 and at2 which contain the same valid access token.
Open a connection with pooling enabled (which is the default); set SqlConnection.AccessToken to at1 and run a query. Close the connection. Open a second connection with pooling enabled; set SqlConnection.AccessToken to at2 and run a second query. Check the number of open connections on the server; it will be 2.

Repeat the experiment except using at1 both times. There will only be a single open connection on the server.

### Expected behavior
Pooled connections using an explicit access token should be shared even if different string objects containing the same value were used to initialize the connection.

### Further technical details
N/A

### PR DESCRIPTION
**Pull request by cheenamalhotra**
- Fixes #438 + tests added
- Modifies tests to dynamically generate Access Token for testing instead of specifying it in `config.json` file.

### CODE REVIEW

File: src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlConnectionPoolKey.cs [hunk 1]
```diff
@@ -63,7 +63,7 @@ internal string AccessToken
         public override bool Equals(object obj)
         {
             SqlConnectionPoolKey key = obj as SqlConnectionPoolKey;
-            return (key != null && _credential == key._credential && ConnectionString == key.ConnectionString && Object.ReferenceEquals(_accessToken, key._accessToken));
+            return (key != null && _credential == key._credential && ConnectionString == key.ConnectionString && _accessToken == key._accessToken);
```
Comment (saurabh500) [inline src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlConnectionPoolKey.cs]: Can you do a ordinal comparison here for AccessToken ?

Comment (cheenamalhotra) [reply src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlConnectionPoolKey.cs]: Sure, I just kept it similar to other comparisons.
Should I change that for Access Token only or for other properties too?

Comment (saurabh500) [reply src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlConnectionPoolKey.cs]: Access Token only.

File: src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlConnectionPoolKey.cs [hunk 2]
```diff
@@ -63,7 +63,10 @@ internal string AccessToken
         public override bool Equals(object obj)
         {
             SqlConnectionPoolKey key = obj as SqlConnectionPoolKey;
-            return (key != null && _credential == key._credential && ConnectionString == key.ConnectionString && _accessToken == key._accessToken);
+            return (key != null
+                && _credential == key._credential
+                && string.CompareOrdinal(ConnectionString, key.ConnectionString) == 0
```
Comment (saurabh500) [inline src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlConnectionPoolKey.cs]: Dont change the check for ConnectionString. Only Access Token

File: src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlConnectionPoolKey.cs [hunk 3]
```diff
@@ -63,7 +63,10 @@ internal string AccessToken
         public override bool Equals(object obj)
         {
             SqlConnectionPoolKey key = obj as SqlConnectionPoolKey;
-            return (key != null && _credential == key._credential && ConnectionString == key.ConnectionString && Object.ReferenceEquals(_accessToken, key._accessToken));
+            return (key != null
+                && _credential == key._credential
+                && ConnectionString == key.ConnectionString
+                && string.CompareOrdinal(_accessToken, key._accessToken) == 0);
```
Comment (saurabh500) [inline src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlConnectionPoolKey.cs]: I forgot to ask, what about NetFx ?

Comment (saurabh500) [reply src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlConnectionPoolKey.cs]: @cheenamalhotra

Comment (saurabh500) [reply src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlConnectionPoolKey.cs]: Never mind. I see it

**Pull request merged**